webapi/CopilotChat/Controllers/ChatMemoryController.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.SemanticKernel.Memory;
+using SemanticKernel.Service.CopilotChat.Options;
+using SemanticKernel.Service.CopilotChat.Skills.ChatSkills;
+using SemanticKernel.Service.CopilotChat.Storage;
+
+namespace SemanticKernel.Service.CopilotChat.Controllers;
+
+/// <summary>
+/// Controller for getting chat session semantic memory data.
+/// </summary>
+[ApiController]
+[Authorize]
+public class ChatMemoryController : ControllerBase
+{
+    private readonly ILogger<ChatMemoryController> _logger;
+
+    private readonly PromptsOptions _promptOptions;
+
+    private readonly ChatSessionRepository _chatSessionRepository;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatMemoryController"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="promptsOptions">The prompts options.</param>
+    /// <param name="chatParticipantRepository">The chat participant repository.</param>
+    public ChatMemoryController(
+        ILogger<ChatMemoryController> logger,
+        IOptions<PromptsOptions> promptsOptions,
+        ChatSessionRepository chatSessionRepository)
+    {
+        this._logger = logger;
+        this._promptOptions = promptsOptions.Value;
+        this._chatSessionRepository = chatSessionRepository;
+    }
+
+    /// <summary>
+    /// Join a use to a chat session given a chat id and a user id.
+    /// </summary>
+    /// <param name="messageRelayHubContext">Message Hub that performs the real time relay service.</param>
+    /// <param name="chatParticipantParam">Contains the user id and chat id.</param>
+    [HttpGet]
+    [Route("chatMemory/{chatId:guid}/{memoryName}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> GetSemanticMemoriesAsync(
+        [FromServices] ISemanticTextMemory semanticTextMemory,
+        [FromRoute] string chatId,
+        [FromRoute] string memoryName)
+    {
+        // Make sure the chat session exists.
+        if (!await this._chatSessionRepository.TryFindByIdAsync(chatId, v => _ = v))
+        {
+            return this.BadRequest("Chat session does not exist.");
```

Comment:
nit: for debuggability I think it would be helpful to put the provided chatId in the response: $"Chat session with id {chatId} does not exist."

Same comment for the memoryName check below.